src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Nls.cs

```diff
@@ -317,26 +317,46 @@ private unsafe int NlsIndexOfCore(ReadOnlySpan<char> source, ReadOnlySpan<char>
             return FindString(positionFlag | (uint)GetNativeCompareFlags(options), source, target, matchLengthPtr);
         }
 
-        private unsafe bool NlsStartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options)
+        private unsafe bool NlsStartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options, int* matchLengthPtr)
         {
             Debug.Assert(!GlobalizationMode.Invariant);
             Debug.Assert(GlobalizationMode.UseNls);
 
             Debug.Assert(!prefix.IsEmpty);
             Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
 
-            return FindString(FIND_STARTSWITH | (uint)GetNativeCompareFlags(options), source, prefix, null) >= 0;
+            int idx = FindString(FIND_STARTSWITH | (uint)GetNativeCompareFlags(options), source, prefix, matchLengthPtr);
+            if (idx >= 0)
```

Comment:
Yup, this has always been the case with NLS. Here's a simple demo:

```cpp
#include <iostream>
#include <Windows.h>

int main()
{
    int chFound;
    int index = FindNLSStringEx(L"", FIND_STARTSWITH | NORM_IGNORESYMBOLS,
        L"''Hello there!", -1,
        L"Hello", -1,
        &chFound,
        nullptr, nullptr, 0);

    std::cout << index << "\n";
    std::cout << chFound << "\n";
}
```

This will print (2, 5) instead of (0, 7). In our `StartsWith` logic, since we return a simple bool instead of the actual index, we assume the caller wants the "matchLength" value normalized to 7 (so that the length is correct assuming a 0 index).